### PR TITLE
feat: add `sortBy` function

### LINKED ISSUE #74: Add `sortBy` function
# Suggestion

## 💻 Use Cases
```ts
const products = [
  { id: 1, name: 'T-shirt', price: 50 },
  { id: 2, name: 'Phone case', price: 30 },
  { id: 3, name: 'Key ring', price: 15 },
];
sortBy(a => a.price, products);
```

### PR DESCRIPTION
**Pull request by ppeeou**
Fixes #74

### CODE REVIEW

File: src/sortBy.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * // [{id:2,name:'bar'}, {id:3, name:'lee'}, {id:4,name:'foo'}]
+ * ```
+ */
+// prettier-ignore
+function sortBy(
+    f: (a: any) => unknown,
+    iterable: readonly []
+): any[];
+
+// prettier-ignore
```
Comment (hg-pyun) [inline src/sortBy.ts:25]: Why did you do `prettier-ignore`?

File: src/sortBy.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  | IterableInfer<T>[]
+  | Promise<IterableInfer<T>[]>
+  | ((iterable: T) => ReturnValueType<T, IterableInfer<T>[]>) {
+  if (iterable === undefined) {
+    return (iterable: T) => {
+      return sortBy(f, iterable as any) as ReturnValueType<
+        T,
+        IterableInfer<T>[]
+      >;
+    };
```
Comment (hg-pyun) [inline src/sortBy.ts:53]: How about this?
```suggestion
    return (iterable: T) => sortBy(f, iterable as any) as ReturnValueType<
        T,
        IterableInfer<T>[]
      >
```

File: test/sortBy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import { filter, identity, pipe, sortBy, toAsync } from "../src/index";
+
+describe("sort", function () {
```
Comment (shine1594) [inline test/sortBy.spec.ts:3]:
```suggestion
describe("sort", function () {
```

File: src/sortBy.ts (hunk 1 above)
Comment (ppeeou) [reply src/sortBy.ts:25]: oh ..  It's not necessary. Thank you for comments

File: test/sortBy.spec.ts (hunk 3 above)
Comment (ppeeou) [reply test/sortBy.spec.ts]: I changed it :)

**Review by shine1594 — APPROVED** [review]

**Review by hg-pyun — APPROVED** [review]
Looks good to me

**Pull request merged**